core/dir/shared/glob.rb

```diff
@@ -49,6 +49,17 @@
       result.should == result.sort
     end
 
+    it "result is sorted with any non false value of sort:" do
+      result = Dir.send(@method, '*', sort: 0)
+      result.should == result.sort
+
+      result = Dir.send(@method, '*', sort: nil)
```

Comment:
This one feels like a bug worth reporting to https://bugs.ruby-lang.org/projects/ruby-master/issues, could you do that?
`nil` is normally always considered `false` or have separate semantics in Ruby.

Reply:
Yes, was a bit confused that `nil` doesn't change sorting, but at the same time, false feels like the only correct way to say – don't sort it!

Will report and attach the patch file